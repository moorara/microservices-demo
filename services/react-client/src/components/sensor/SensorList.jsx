import React from 'react'
import PropTypes from 'prop-types'

import SensorListItem from './SensorListItem'

// Presentational Component
const SensorList = ({ sensors }) => (
  <table className="table is-bordered is-striped is-fullwidth is-hoverable">
    <thead>
      <tr>
        <th>Name</th>
        <th>Unit</th>
        <th>Minimum Safe Value</th>
        <th>Maximum Safe Value</th>
      </tr>
    </thead>
    <tbody>
      {sensors.map(sensor => <SensorListItem key={sensor.id} sensor={sensor} />)}
    </tbody>
  </table>
)

SensorList.propTypes = {
  sensors: PropTypes.array.isRequired
}

export default SensorList
