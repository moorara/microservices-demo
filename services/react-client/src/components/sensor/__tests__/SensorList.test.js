import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import SensorList from '../SensorList'

describe('SensorList', () => {
  const props = {
    sensors: []
  }

  test('renders react component with props', () => {
    props.sensors = [
      { id: 'aaaa-aaaa', siteId: '1111-1111', name: 'temperature', unit: 'celsius', minSafe: -30, maxSafe: 30 },
      { id: 'bbbb-bbbb', siteId: '2222-2222', name: 'temperature', unit: 'fahrenheit', minSafe: -22, maxSafe: 86 }
    ]
    const wrapper = shallow(<SensorList {...props} />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
    expect(wrapper.find('thead tr th').at(0).text()).toBe('Name')
    expect(wrapper.find('thead tr th').at(1).text()).toBe('Unit')
    expect(wrapper.find('thead tr th').at(2).text()).toBe('Minimum Safe Value')
    expect(wrapper.find('thead tr th').at(3).text()).toBe('Maximum Safe Value')
    expect(wrapper.find('SensorListItem').get(0).props.sensor).toEqual(props.sensors[0])
    expect(wrapper.find('SensorListItem').get(1).props.sensor).toEqual(props.sensors[1])
  })
  test('renders react component with props', () => {
    props.sensors = [
      { id: 'cccc-cccc', siteId: '3333-3333', name: 'pressure', unit: 'pascal', minSafe: 50000, maxSafe: 100000 },
      { id: 'dddd-dddd', siteId: '4444-4444', name: 'pressure', unit: 'atmosphere', minSafe: 0.5, maxSafe: 1 }
    ]
    const wrapper = shallow(<SensorList {...props} />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
    expect(wrapper.find('thead tr th').at(0).text()).toBe('Name')
    expect(wrapper.find('thead tr th').at(1).text()).toBe('Unit')
    expect(wrapper.find('thead tr th').at(2).text()).toBe('Minimum Safe Value')
    expect(wrapper.find('thead tr th').at(3).text()).toBe('Maximum Safe Value')
    expect(wrapper.find('SensorListItem').get(0).props.sensor).toEqual(props.sensors[0])
    expect(wrapper.find('SensorListItem').get(1).props.sensor).toEqual(props.sensors[1])
  })
})
