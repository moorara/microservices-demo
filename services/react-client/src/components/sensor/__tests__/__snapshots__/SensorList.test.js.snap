// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SensorList renders react component with props 1`] = `
<table
  className="table is-bordered is-striped is-fullwidth is-hoverable"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Unit
      </th>
      <th>
        Minimum Safe Value
      </th>
      <th>
        Maximum Safe Value
      </th>
    </tr>
  </thead>
  <tbody>
    <SensorListItem
      key="aaaa-aaaa"
      sensor={
        Object {
          "id": "aaaa-aaaa",
          "maxSafe": 30,
          "minSafe": -30,
          "name": "temperature",
          "siteId": "1111-1111",
          "unit": "celsius",
        }
      }
    />
    <SensorListItem
      key="bbbb-bbbb"
      sensor={
        Object {
          "id": "bbbb-bbbb",
          "maxSafe": 86,
          "minSafe": -22,
          "name": "temperature",
          "siteId": "2222-2222",
          "unit": "fahrenheit",
        }
      }
    />
  </tbody>
</table>
`;

exports[`SensorList renders react component with props 2`] = `
<table
  className="table is-bordered is-striped is-fullwidth is-hoverable"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Unit
      </th>
      <th>
        Minimum Safe Value
      </th>
      <th>
        Maximum Safe Value
      </th>
    </tr>
  </thead>
  <tbody>
    <SensorListItem
      key="cccc-cccc"
      sensor={
        Object {
          "id": "cccc-cccc",
          "maxSafe": 100000,
          "minSafe": 50000,
          "name": "pressure",
          "siteId": "3333-3333",
          "unit": "pascal",
        }
      }
    />
    <SensorListItem
      key="dddd-dddd"
      sensor={
        Object {
          "id": "dddd-dddd",
          "maxSafe": 1,
          "minSafe": 0.5,
          "name": "pressure",
          "siteId": "4444-4444",
          "unit": "atmosphere",
        }
      }
    />
  </tbody>
</table>
`;
